### ISSUE TITLE
Pressable unnecessarily re-renders if `disabled` prop is true

### ISSUE THREAD

**Issue by Saadnajmi** [facebook/react-native#34986]
### Description

See: https://github.com/facebook/react-native/blob/b55518c4bf22ea6f409b58151d9c84fa161d6e02/Libraries/Components/Pressable/Pressable.js#L206 

Based in whether you pass `disabled` into Pressable, Pressable will conditionally create a new object as a prop for its underlying `<View>`. Because of that, it will fail React's shallow-compare reconciliation step and unnecessarily re-render.

We could fix this by moving the calculation of AccessibilityState into a `useMemo` call, such as the one used for the Pressability config a few lines below. As far as I can tell, this issue still exists on 0.70+ / main. Or a new `useMemo` call. I'll look into what a good fix can be.

### Version

0.68 

### Output of `npx react-native info`

System:
    OS: macOS 13.0
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 43.46 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.20.0 - ~/.nvm/versions/node/v14.20.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 6.14.17 - ~/.nvm/versions/node/v14.20.0/bin/npm
    Watchman: 2022.03.21.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.0, iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.3, 30.0.1, 30.0.2, 30.0.3, 31.0.0
      System Images: android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8193401
    Xcode: 14.0/14A5294g - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Simply put <Pressable disabled> in your app anywhere.
Observe that on a re-render, `<Pressable disabled>` will also re-render, despite none of the props changing. 

### Snack, code example, screenshot, or link to a repository

I first noticed it with a failing jest test while  working on integrating Pressable in our own repository FluentUI React Native: https://github.com/microsoft/fluentui-react-native/pull/2153

I can try and create a minimal snack that counts/shows the re-renders.

**Comment by Saadnajmi** [comment]
Confirmed this is an issue despite the `React.memo` call
